### ISSUE TITLE
velocity is not a constant number

### ISSUE THREAD

**Issue by 285639863** [R0S3NB3RG/PINNs-for-wave-equation#1]
I saw that the velocity is a constant 1 in your code. I want to use a more complex velocity model to replace this constant velocity model for training, such as the marmousi velocity model, but I don't know how to set it up because this velocity requires the use of a tensor related to the x and y coordinates of space.Do you have some ideas about this question?
Best wishes!

**Comment by R0S3NB3RG** [comment]
Hi,
Thanks for your email. However, I’m on my way these days. I will give my suggestion as soon as possible.
Cheers.



发自我的iPhone


------------------ Original ------------------
From: 285639863 ***@***.***&gt;
Date: Mon,Dec 18,2023 7:45 AM
To: R0S3NB3RG/PINNs-for-wave-equation ***@***.***&gt;
Cc: Subscribed ***@***.***&gt;
Subject: Re: [R0S3NB3RG/PINNs-for-wave-equation] velocity is not a constant number(Issue #1)





 
I saw that the velocity is a constant 1 in your code. I want to use a more complex velocity model to replace this constant velocity model for training, such as the marmousi velocity model, but I don't know how to set it up because this velocity requires the use of a tensor related to the x and y coordinates of space.Do you have some ideas about this question?
 Best wishes!

**Comment by R0S3NB3RG** [comment]
Hi,
I think you can write a function for your velocity like velocity(x,t), which will return the the velocity at the specific point. When you doing the loss function part, use this return value to construct the relationship between your inputs and outputs.
And my friend provide another creative way, which is to use the&nbsp;Lagrangian L(q,q_dot) which is a function depends on general position and general velocity and use Euler-Lagrangian Equation and least action to construct your loss function. The main challenge for this method is how to get the Lagrangian based on your PDE.
(In your writing style, it seems that you are not a English speaker. If you are a Chinese, we can discuss it on Wechat if you like. If not, just forget about it)
Cheers!


Rosen
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "R0S3NB3RG/PINNs-for-wave-equation"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月18日(星期一) 下午3:45
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[R0S3NB3RG/PINNs-for-wave-equation] velocity is not a constant number (Issue #1)





 
I saw that the velocity is a constant 1 in your code. I want to use a more complex velocity model to replace this constant velocity model for training, such as the marmousi velocity model, but I don't know how to set it up because this velocity requires the use of a tensor related to the x and y coordinates of space.Do you have some ideas about this question?
 Best wishes!

**Comment by 285639863** [comment]
Thanks a lot！I have solved it using your method with a velocity function!



dreamland
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "R0S3NB3RG/PINNs-for-wave-equation"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月20日(星期三) 中午12:00
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [R0S3NB3RG/PINNs-for-wave-equation] velocity is not a constant number (Issue #1)





 Hi, 
 I think you can write a function for your velocity like velocity(x,t), which will return the the velocity at the specific point. When you doing the loss function part, use this return value to construct the relationship between your inputs and outputs. 
 And my friend provide another creative way, which is to use the&amp;nbsp;Lagrangian L(q,q_dot) which is a function depends on general position and general velocity and use Euler-Lagrangian Equation and least action to construct your loss function. The main challenge for this method is how to get the Lagrangian based on your PDE. 
 (In your writing style, it seems that you are not a English speaker. If you are a Chinese, we can discuss it on Wechat if you like. If not, just forget about it) 
 Cheers! 
 
 
 Rosen 
 ***@***.*** 
 
 
 
 &amp;nbsp; 
 
 
 
 
 ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 
 发件人:                                                                                                                        "R0S3NB3RG/PINNs-for-wave-equation"                                                                                    ***@***.***&amp;gt;; 
 发送时间:&amp;nbsp;2023年12月18日(星期一) 下午3:45 
 ***@***.***&amp;gt;; 
 ***@***.***&amp;gt;; 
 主题:&amp;nbsp;[R0S3NB3RG/PINNs-for-wave-equation] velocity is not a constant number (Issue #1) 
 
 
 
 
 
   
 I saw that the velocity is a constant 1 in your code. I want to use a more complex velocity model to replace this constant velocity model for training, such as the marmousi velocity model, but I don't know how to set it up because this velocity requires the use of a tensor related to the x and y coordinates of space.Do you have some ideas about this question? 
  Best wishes!
